plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/RewriteStringComparison.java

```diff
@@ -70,19 +70,48 @@ public Optional<ParameterizedExpression> rewrite(Call expression, Captures captu
         Variable firstArgument = captures.get(FIRST_ARGUMENT);
         Variable secondArgument = captures.get(SECOND_ARGUMENT);
 
-        if (isClob(firstArgument, context) || isClob(secondArgument, context)) {
-            return Optional.empty();
+        boolean[] clobInfo = {false, false};
+        if (isClob(firstArgument, context)) {
+            clobInfo[0] = true;
+        }
+        if (isClob(secondArgument, context)) {
+            clobInfo[1] = true;
         }
         return context.defaultRewrite(firstArgument).flatMap(first ->
                 context.defaultRewrite(secondArgument).map(second ->
                         new ParameterizedExpression(
-                                "(%s) %s (%s)".formatted(first.expression(), comparison.getOperator(), second.expression()),
+                                buildComparison(first.expression(), comparison.getOperator(), second.expression(), clobInfo),
                                 ImmutableList.<QueryParameter>builder()
                                         .addAll(first.parameters())
                                         .addAll(second.parameters())
                                         .build())));
     }
 
+    private static String buildComparison(
+            String firstExpression,
+            String operator,
+            String secondExpression,
+            boolean[] clobInfo)
+    {
+        if (clobInfo[0] || clobInfo[1]) {
+            String firstArgument = firstExpression;
+            String secondArgument = secondExpression;
+            if (!clobInfo[0]) {
+                firstArgument = toNClob(firstExpression);
+            }
+            if (!clobInfo[1]) {
+                secondArgument = toNClob(secondExpression);
+            }
+            return "DBMS_LOB.compare(%s, %s) %s 0".formatted(firstArgument, secondArgument, operator);
+        }
+        return "(%s) %s (%s)".formatted(firstExpression, operator, secondExpression);
+    }
+
+    private static String toNClob(String expression)
+    {
+        return "TO_NCLOB(%s)".formatted(expression);
```

Comment:
What if during trino reads the remote DB the unsupported by trino type (some remote-specific type) was force-mapped to text, 
later here it was capturedAs CharType or VarcharType ?
Will TO_NCLOB work correctly? 
Maybe it's the case here too https://github.com/trinodb/trino/pull/22203/files#diff-510ce28e8ade87630227de284298df1cb44186f73bc9f2219e1f6ad6505d410aR66

Reply:
removed this change. `nclob` cast pushdown is out of scope of this PR.